Nodejs/Product/TestAdapter/TestDiscovererWorker.cs

```diff
@@ -26,14 +27,16 @@ public TestDiscovererWorker(string testSource, string nodeExePath)
             this.nodeExePath = nodeExePath;
         }
 
-        public void DiscoverTests(string testFolderPath, TestFramework testFx, IMessageLogger logger, ITestCaseDiscoverySink discoverySink)
+        public void DiscoverTests(string testFolderPath, TestFramework testFx, IMessageLogger logger, ITestCaseDiscoverySink discoverySink, string testDiscovererName)
         {
             var fileList = Directory.EnumerateFiles(testFolderPath, "*.js", SearchOption.AllDirectories).Where(x => !x.Contains(NodejsConstants.NodeModulesFolder));
-            this.DiscoverTests(fileList, testFx, logger, discoverySink);
+            this.DiscoverTests(fileList, testFx, logger, discoverySink, testDiscovererName);
         }
 
-        public void DiscoverTests(IEnumerable<string> fileList, TestFramework testFx, IMessageLogger logger, ITestCaseDiscoverySink discoverySink)
+        public void DiscoverTests(IEnumerable<string> fileList, TestFramework testFx, IMessageLogger logger, ITestCaseDiscoverySink discoverySink, string testDiscovererName)
         {
+            TelemetryHelper.LogTestDiscoveryStarted(testFx.Name, testDiscovererName);
```

Comment:
The framework name comes from a fixed list we control?

Reply:
The name comes from a list of directories we expect to exists within the VS installation. Technically, the user might be able to add more directories.

Sounds like we might need to fix to specific names?

Reply:
Life would be simpler if we didn't report any user-specified strings, yes.  If that's complicated, grab me and we'll figure something out.

Reply:
I have added a simple whitelist before calling the method. Hope that's enough.